### PR TITLE
Pinning qol

### PR DESCRIPTION
**Pull request by wiviw5**
Allows thread owners to pin and unpin messages within their own threads via reacting with the 📌 emoji.

### CODE REVIEW

File: src/main/java/net/hypixel/nerdbot/listener/ActivityListener.java [hunk 1]
```diff
@@ -198,6 +199,33 @@ public void onReactionReceived(MessageReactionAddEvent event) {
...
+
+            if (!(event.getGuildChannel().asThreadChannel().getOwnerId().equals(discordUser.getDiscordId()))){
+                return; // Ignoring IDs that are not from the owner of the thread.
+            }
+
+            // We get the message we are checking to pin or unpin it.
+            Message message = event.retrieveMessage().complete();
+            if (message.isPinned()){
+                // Since it is pinned, we unpin it here and remove all :pushpin: emojis.
+                message.unpin().queue();
```
Comment (Aerhhh) [inline src/main/java/net/hypixel/nerdbot/listener/ActivityListener.java:218]: Change this to `complete()`

File: src/main/java/net/hypixel/nerdbot/listener/ActivityListener.java [hunk 2]
```diff
@@ -198,6 +199,33 @@ public void onReactionReceived(MessageReactionAddEvent event) {
...
+                message.clearReactions(pushpin).queue();
+            } else {
+                // It has not been pinned yet, we pin it and remove the users reaction while adding our own.
+                message.pin().queue();
```
Comment (Aerhhh) [inline src/main/java/net/hypixel/nerdbot/listener/ActivityListener.java:222]: Change this to `complete()`

File: src/main/java/net/hypixel/nerdbot/listener/ActivityListener.java [hunk 3]
```diff
@@ -198,6 +199,33 @@ public void onReactionReceived(MessageReactionAddEvent event) {
...
 
+        // Here is logic for handling pinning of messages within Threads by the owner. This is done by only the :pushpin: Emoji.
+        if (event.getReaction().getEmoji().getName().equals("\uD83D\uDCCC")) {
+            Emoji pushpin = Emoji.fromUnicode("\uD83D\uDCCC");
+
+            if (!(event.getGuildChannel() instanceof ThreadChannel)){
+                return; // Ignoring any channels not within a Thread channel.
+            }
+
+            if (!(event.getGuildChannel().asThreadChannel().getOwnerId().equals(discordUser.getDiscordId()))){
```
Comment (Aerhhh) [inline src/main/java/net/hypixel/nerdbot/listener/ActivityListener.java:210]:
```suggestion
            if (!event.getGuildChannel().asThreadChannel().getOwnerId().equals(discordUser.getDiscordId())) {

```

File: src/main/java/net/hypixel/nerdbot/listener/ActivityListener.java [hunk 4]
```diff
@@ -198,6 +199,33 @@ public void onReactionReceived(MessageReactionAddEvent event) {
             return; // Ignore Empty User
         }
 
+        // Here is logic for handling pinning of messages within Threads by the owner. This is done by only the :pushpin: Emoji.
+        if (event.getReaction().getEmoji().getName().equals("\uD83D\uDCCC")) {
+            Emoji pushpin = Emoji.fromUnicode("\uD83D\uDCCC");
+
+            if (!(event.getGuildChannel() instanceof ThreadChannel)){
```
Comment (Aerhhh) [inline src/main/java/net/hypixel/nerdbot/listener/ActivityListener.java:206]: Could probably use pattern matching on ThreadChannel and use it further down
Also, formatting pls

Comment (Aerhhh) [reply src/main/java/net/hypixel/nerdbot/listener/ActivityListener.java:206]: Would actually also probably swap this and line 203

**Review by Aerhhh — CHANGES_REQUESTED** [review]

File: src/main/java/net/hypixel/nerdbot/listener/ActivityListener.java (hunk 1 above)
Comment (wiviw5) [reply src/main/java/net/hypixel/nerdbot/listener/ActivityListener.java:218]: Complete, will be pushed in next commit.

File: src/main/java/net/hypixel/nerdbot/listener/ActivityListener.java (hunk 2 above)
Comment (wiviw5) [reply src/main/java/net/hypixel/nerdbot/listener/ActivityListener.java:222]: Complete, will be pushed in next commit.

**Pull request merged**